### PR TITLE
Image lightbox

### PR DESCRIPTION
**Pull request by dstrawberrygirl**
Building on the work started by @kswenson, this PR requires a compatible `lara-interactive-api` in order to work (see https://github.com/concord-consortium/lara/pull/571). 

Images now have a zoom in magnifying glass icon in the bottom right, alternatively the image itself could be clicked to initiate the call to show the image in a lightbox. The LARA side accepts simple html to render the title of the image, so we're passing in the caption and credit for the images that have been authored here.

### CODE REVIEW

File: src/image/components/runtime.tsx [hunk 1]
```diff
@@ -30,9 +40,32 @@ export const Runtime: React.FC<IProps> = ({ authoredState, report }) => {
     }
   };
 
+  const handleClick = () => {
+    const { url, highResUrl, credit, caption, creditLink } = authoredState;
+    const size = imageSize.current?.width && imageSize.current?.height
+                  ? { width: imageSize.current.width, height: imageSize.current.height }
+                  : undefined;
+    const allowUpscale = authoredState.scaling === "fitWidth";
+    const modalImageUrl = highResUrl ? highResUrl : url;
```
Comment (kswenson) [inline src/image/components/runtime.tsx:49]:
```suggestion
    const modalImageUrl = highResUrl || url;
```

File: src/image/components/runtime.tsx [hunk 2]
```diff
@@ -30,9 +40,32 @@ export const Runtime: React.FC<IProps> = ({ authoredState, report }) => {
...
+    modalImageUrl && showModal({ uuid: "image", type: "lightbox", url: modalImageUrl,
+                        isImage: true, size, allowUpscale, title });
+  }
+
+  const getCreditLink = () => {
+    const { creditLink, creditLinkDisplayText } = authoredState;
+    return (
+      <div className={css.creditLink}>
+        <a href={authoredState.creditLink} target="_blank">
+          {authoredState.creditLinkDisplayText ? authoredState.creditLinkDisplayText : authoredState.creditLink}
```
Comment (kswenson) [inline src/image/components/runtime.tsx:60]:
```suggestion
          {authoredState.creditLinkDisplayText || authoredState.creditLink}
```

File: src/image/components/runtime.tsx [hunk 3]
```diff
@@ -30,9 +40,32 @@ export const Runtime: React.FC<IProps> = ({ authoredState, report }) => {
...
+    const allowUpscale = authoredState.scaling === "fitWidth";
+    const modalImageUrl = highResUrl ? highResUrl : url;
+    const title = `<strong>${caption || ""}</strong> <em>${credit || ""}</em> ${creditLink && ReactDOMServer.renderToString(getCreditLink())}`;
+    modalImageUrl && showModal({ uuid: "image", type: "lightbox", url: modalImageUrl,
+                        isImage: true, size, allowUpscale, title });
+  }
+
+  const getCreditLink = () => {
+    const { creditLink, creditLinkDisplayText } = authoredState;
+    return (
```
Comment (kswenson) [inline src/image/components/runtime.tsx:57]:
```suggestion
    return creditLink && (
```
If you handle the null/undefined case here, clients wouldn't need to test for it.

File: src/image/components/runtime.tsx [hunk 4]
```diff
@@ -30,9 +40,32 @@ export const Runtime: React.FC<IProps> = ({ authoredState, report }) => {
...
 
+  const handleClick = () => {
+    const { url, highResUrl, credit, caption, creditLink } = authoredState;
+    const size = imageSize.current?.width && imageSize.current?.height
+                  ? { width: imageSize.current.width, height: imageSize.current.height }
+                  : undefined;
+    const allowUpscale = authoredState.scaling === "fitWidth";
+    const modalImageUrl = highResUrl ? highResUrl : url;
+    const title = `<strong>${caption || ""}</strong> <em>${credit || ""}</em> ${creditLink && ReactDOMServer.renderToString(getCreditLink())}`;
+    modalImageUrl && showModal({ uuid: "image", type: "lightbox", url: modalImageUrl,
```
Comment (kswenson) [inline src/image/components/runtime.tsx:51]: Forgot to check this earlier -- you need to generate a real UUID here.

Comment (dstrawberrygirl) [reply src/image/components/runtime.tsx]: Thanks - added.

File: src/image/components/runtime.tsx (hunk 1 above)
Comment (dstrawberrygirl) [reply src/image/components/runtime.tsx]: I keep forgetting this syntax, thanks!

File: src/image/components/runtime.tsx [hunk 5]
```diff
@@ -30,9 +41,33 @@ export const Runtime: React.FC<IProps> = ({ authoredState, report }) => {
...
+    const title = `<strong>${caption || ""}</strong> <em>${credit || ""}</em> ${ReactDOMServer.renderToString(getCreditLink() || <span/>)}`;
+    modalImageUrl && showModal({ uuid, type: "lightbox", url: modalImageUrl,
+                        isImage: true, size, allowUpscale, title });
+  }
+
+  const getCreditLink = () => {
+    const { creditLink, creditLinkDisplayText } = authoredState;
+    return creditLink && (
+      <div className={css.creditLink}>
+        <a href={creditLink} target="_blank">
```
Comment (kswenson) [inline src/image/components/runtime.tsx:61]:
```suggestion
        <a href={creditLink} target="_blank" rel="noopener">
```
for [security reasons](https://web.dev/external-anchors-use-rel-noopener/).

**Pull request merged**